src/sentry/monitors/serializers.py

```diff
@@ -6,22 +6,68 @@
 from sentry.api.serializers import ProjectSerializerResponse, Serializer, register, serialize
 from sentry.models import Project
 
-from .models import Monitor, MonitorCheckIn
+from .models import Monitor, MonitorCheckIn, MonitorEnvironment
+
+
+@register(MonitorEnvironment)
+class MonitorEnvironmentSerializer(Serializer):
+    def serialize(self, obj, attrs, user):
+        return {
+            "monitor_id": obj.monitor.id,
+            "status": obj.get_status_display(),
+            "name": obj.environment.name,
+            "lastCheckIn": obj.last_checkin,
+            "nextCheckIn": obj.next_checkin,
+            "dateCreated": obj.monitor.date_added,
+        }
+
+
+class MonitorEnvironmentSerializerResponse(TypedDict):
+    monitor_id: str
+    name: str
+    status: str
+    dateCreated: datetime
+    lastCheckIn: datetime
+    nextCheckIn: datetime
 
 
 @register(Monitor)
 class MonitorSerializer(Serializer):
-    def get_attrs(self, item_list, user):
+    def __init__(self, environments=None):
+        self.environments = environments
+
+    def get_attrs(self, item_list, user, **kwargs):
         # TODO(dcramer): assert on relations
         projects = {
-            d["id"]: d
-            for d in serialize(
+            p["id"]: p
+            for p in serialize(
                 list(Project.objects.filter(id__in=[i.project_id for i in item_list])), user
             )
         }
 
+        environment_data = {}
+        if self.environments:
+            monitor_environments = {
+                monitor_environment.pop("monitor_id"): {
+                    monitor_environment.pop("name"): monitor_environment
```

Comment:
It might be a good idea to not `.pop` here. The data will be duplicated, but it might be easier to work on in the FE to just have the whole serialized `monitor_environment`

Reply:
yeah...I think it's kind of ugly on the API side of things. will ping them

Reply:
will make this a list & keep the name in the object